packages/flutter_migrate/lib/src/compute.dart

```diff
@@ -0,0 +1,1045 @@
+// found in the LICENSE file.
+
+import 'base/common.dart';
+import 'base/file_system.dart';
+import 'base/logger.dart';
+import 'base/project.dart';
+import 'custom_merge.dart';
+import 'environment.dart';
+import 'flutter_project_metadata.dart';
+import 'migrate_logger.dart';
+import 'result.dart';
+import 'utils.dart';
+
+// This defines paths of files and directories relative to the project root
+// that should be ignored by the migrate tool regardless of .gitignore and
+// config settings.
+// Paths use `/` as a stand-in for path separator.
+const List<String> _skippedFiles = <String>[
+  'ios/Runner.xcodeproj/project.pbxproj', // Xcode managed configs that may not merge cleanly.
+  'README.md', // changes to this shouldn't be overwritten since is is user owned.
+];
+
+const List<String> _skippedDirectories = <String>[
+  '.dart_tool', // The .dart_tool generated dir.
+  '.git', // Git metadata.
+  'assets', // Common directory for user assets.
+  'build', // Build artifacts.
+  'lib', // Always user owned and we don't want to overwrite their apps.
+  'test', // Typically user owned and flutter-side changes are not relevant.
+];
+
+// Returns true for paths relative to the project root that should be skipped
+// completely by the migrate tool.
+bool _skipped(String localPath, FileSystem fileSystem,
+    {Set<String>? skippedPrefixes}) {
+  for (final String path in _skippedFiles) {
+    if (path.replaceAll('/', fileSystem.path.separator) == localPath) {
+      return true;
+    }
+  }
+  if (_skippedFiles.contains(localPath)) {
+    return true;
+  }
+  for (final String dir in _skippedDirectories) {
+    if (localPath.startsWith(
+        '${dir.replaceAll('/', fileSystem.path.separator)}${fileSystem.path.separator}')) {
+      return true;
+    }
+  }
+  if (skippedPrefixes != null) {
+    return skippedPrefixes.any((String prefix) =>
+        localPath.startsWith('$prefix${fileSystem.path.separator}'));
+  }
+  return false;
+}
+
+// File extensions that the tool will not attempt to merge. Changes
+// in files with these extensions will be accepted wholesale.
+//
+// The executables and binaries in this list are not meant to be
+// comprehensive and need only cover the files that are generated
+// in `flutter create` as only files generated by the template
+// will be attempted to be merged.
+const List<String> _doNotMergeFileExtensions = <String>[
+  // Don't merge image files
+  '.bmp',
+  '.gif',
+  '.jpg',
+  '.jpeg',
+  '.png',
+  '.svg',
+  // Don't merge compiled artifacts and executables
+  '.dll',
+  '.exe',
+  '.jar',
+  '.so',
+];
+
+// These files should always go through the migrate process as
+// they are either integral to the mirgate process or we expect
+// new versions of this file to always be desired.
+const Set<String> _alwaysMigrateFiles = <String>{
+  '.metadata', // .metadata tracks key migration information.
+  'android/gradle/wrapper/gradle-wrapper.jar',
```

Comment:
*.jar is part of `_doNotMergeFileExtensions`, does this take precedence?

Reply:
Yes. The _doNotMerge list only controls what will be attempted to be git-merged. This list will blindly accept the new version, which does not pass the merge step.

Reply:
SGTM